docs/CpuSamplingEngines.md

```diff
@@ -0,0 +1,47 @@
+# CPU Sampling Engines
+
+Async-profiler has three options for CPU profiling: `-e cpu`, `-e itimer` and `-e ctimer`.
+
+## cpu & itimer mode
+
+Both cpu and itimer mode measure the CPU time spent by the running threads. For example, 
+if an application uses 2 cpu cores, each with 30% utilization, and the sampling interval is 
+10ms, then the profiler will collect about 2 * 0.3 * 100 = 60 samples per second.
+
+In other words, 1 profiling sample means that one CPU was actively running for N nanoseconds, 
+where N is the profiling interval.
+
+- `itimer` mode is based on [setitimer(ITIMER_PROF)](https://man7.org/linux/man-pages/man2/setitimer.2.html)
+syscall, which ideally generates a signal every given interval of the CPU time consumed by the process.
+- `cpu` mode relies on [perf_events](https://man7.org/linux/man-pages/man2/perf_event_open.2.html). 
+The idea is the same - to generate a signal every `N` nanoseconds of CPU time, which in this case 
+is achieved by configuring PMU to generate an interrupt every `K` CPU cycles.
+
+Ideally, both `itimer` and `cpu` should collect the same number of samples. Typically, the 
+profiles indeed look very similar. However, in [some cases](https://github.com/golang/go/issues/14434)
+cpu profile appears a bit more accurate though, since the signal is delivered exactly to the thread
+that overflowed a hardware counter.
+
+## ctimer mode
+
+[perf_events](https://man7.org/linux/man-pages/man2/perf_event_open.2.html) are not always available,
```

Comment:
I have recently changed behavior of `cpu` option: https://github.com/async-profiler/async-profiler/issues/1044#issuecomment-2453774949
Worth reflecting in the doc.

Reply:
Addressed in [e6282c8](https://github.com/async-profiler/async-profiler/pull/1029/commits/e6282c83bc0aa6007df310fd5e638e9d4d5ed6f1)